commit/plugin.go

```diff
@@ -164,6 +180,18 @@ func (p *Plugin) Observation(
 		return nil, fmt.Errorf("decode query: %w", err)
 	}
 
+	var discoveryObs dt.Observation
+	if p.discoveryProcessor != nil {
+		discoveryObs, err = p.discoveryProcessor.Observation(ctx, dt.Outcome{}, dt.Query{})
+		if err != nil {
+			p.lggr.Errorw("failed to discover contracts", "err", err)
+		}
+		if !p.contractsInitialized {
```

Comment:
If contract addresses change, is there a risk of having stale addresses?

Reply:
Contract addresses won't change but we could have new chains added while the plugin is running.

However, the discovery processor continuously discovers new addresses at the moment, so these new chains get picked up. There's also a test for this in chainlink.

Reply:
A new address would be a problem because we don't unbind the old one and currently ensure that there is only one bound address. This is something we could support pretty easily if we need to in the future.